lib/reverse_sentence.rb

```diff
@@ -1,6 +1,18 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(N)
+# Space complexity: O(N)
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  if my_sentence == nil
+    return nil 
+  else 
+  my_arr = my_sentence.split(/(\s+)(?=(?:[^"]*"[^"]*")*[^"]*$)/) #regex expression from https://stackoverflow.com/questions/8162444/ruby-regex-extracting-words
+  p my_arr
+  joined = []
+  my_arr.each do |word|
+    joined.prepend word
+  end
```

Comment:
Just a note that `.prepend` to an array is an O(n) operation as it has to shift each element down one index.